File: backend/src/main/java/ma/saharatours/service/AuthService.java
package ma.saharatours.service;

import lombok.RequiredArgsConstructor;
import ma.saharatours.dto.AuthResponse;
import ma.saharatours.dto.LoginRequest;
import ma.saharatours.dto.RegisterRequest;
import ma.saharatours.entity.User;
import ma.saharatours.enums.Role;
import ma.saharatours.repository.UserRepository;
import ma.saharatours.security.JwtUtil;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtUtil jwtUtil;
    private final AuthenticationManager authManager;

    public AuthResponse login(LoginRequest request) {
        authManager.authenticate(
                new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword())
        );
        User user = userRepository.findByEmail(request.getEmail())
                .orElseThrow(() -> new RuntimeException("Utilisateur introuvable"));
        String token = jwtUtil.generateToken(user);
        return toResponse(token, user);
    }

    public AuthResponse register(RegisterRequest request) {
        if (userRepository.existsByEmail(request.getEmail())) {
            throw new RuntimeException("Cet email est déjà utilisé");
        }
        User user = User.builder()
                .name(request.getName())
                .email(request.getEmail())
                .password(passwordEncoder.encode(request.getPassword()))
                .role(Role.CLIENT)
                .build();
        userRepository.save(user);
        String token = jwtUtil.generateToken(user);
        return toResponse(token, user);
    }

    private AuthResponse toResponse(String token, User user) {
        return new AuthResponse(
                token,
                user.getId(),
                user.getName(),
                user.getEmail(),
                user.getRole().name(),
                user.getCreatedAt().toString()
        );
    }
}

File: frontend/src/services/authService.js
import api from './api'

export const authService = {
  async login(email, password) {
    const { data } = await api.post('/auth/login', { email, password })
    // Backend returns flat: { token, id, name, email, role, createdAt }
    // Reshape to { token, user } expected by AuthContext
    return {
      token: data.token,
      user: { id: data.id, name: data.name, email: data.email, role: data.role, createdAt: data.createdAt },
    }
  },

  async register({ name, email, password }) {
    const { data } = await api.post('/auth/register', { name, email, password })
    return {
      token: data.token,
      user: { id: data.id, name: data.name, email: data.email, role: data.role, createdAt: data.createdAt },
    }
  },
}
